test/e2e/specs/editor/various/list-view.spec.js

```diff
@@ -0,0 +1,345 @@
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Select the image block in List View.
+		await pageUtils.pressKeyTimes( 'ArrowUp', 2 );
+		await expect(
+			page
+				.getByRole( 'gridcell', {
+					name: 'Image link',
+				} )
+				.locator( 'a' )
+		).toBeFocused();
+		await page.keyboard.press( 'Enter' );
+
+		// Remove the Image block via its options menu in List View.
+		await page
+			.getByRole( 'button', { name: 'Options for Image block' } )
+			.click();
+		await page.getByRole( 'menuitem', { name: /Remove Image/i } ).click();
+
+		// Heading block should be selected as previous block.
+		await expect(
+			page.getByRole( 'document', {
+				name: 'Block: Heading',
+			} )
+		).toBeFocused();
+	} );
+
+	/**
+	 * When all the blocks gets removed from the editor, it inserts a default
+	 * paragraph block; make sure that paragraph block gets selected after
+	 * removing blocks from ListView.
+	 */
+	test( 'selects the default paragraph block after removing all blocks', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Select the Image block as well.
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowUp' );
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Image link',
+				selected: true,
+			} )
+			.waitFor();
+
+		// Remove both blocks.
+		await page
+			.getByRole( 'button', { name: 'Options for Image block' } )
+			.click();
+		await page.getByRole( 'menuitem', { name: /Remove blocks/i } ).click();
+
+		// Newly created paragraph block should be selected.
+		await expect(
+			page.getByRole( 'document', { name: /Empty block/i } )
+		).toBeFocused();
+	} );
+
+	test( 'expands nested list items', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Click first color option from the block placeholder's color picker to
+		// make the inner blocks appear.
+		await page
+			.getByRole( 'button', { name: /Color: /i } )
+			.first()
+			.click();
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Things start off expanded.
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Cover link',
+				expanded: true,
+			} )
+			.waitFor();
+
+		// The child paragraph block should be selected.
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Paragraph link',
+				selected: true,
+			} )
+			.waitFor();
+
+		// Collapse the Cover block.
+		await page
+			.getByRole( 'gridcell', { name: 'Cover link' } )
+			.locator( '.block-editor-list-view__expander[aria-hidden="true"]' )
```

Comment:
Non-blocking: This is where I think `data-testid` will be more explicit in the source code. Right now we don't seem to have any other options though.

Reply:
Cool, I wasn't aware we're using the `testid` approach as well. Should I add it to the expander element?

Reply:
That will be nice! ❤️

Reply:
Agreed! It feels like a good way to go since it allows us to use the `getBy*` format locator. Updated in https://github.com/WordPress/gutenberg/pull/47919/commits/1891141deb51286996eb5127e795a24a4a790320.